src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,50 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.validateAndParseAdParams_(eventData);
+    super.updateInternalOnInit(eventData);
+  }
+
+  /**
+   * Checks validity of ad parameters and rejects with proper message if invalid.
+   * @param eventData an object that contains information details for a particular event
+   *   such as event type, unique Ids, creativeData and environmentData.
+   * @private 
+  */ 
+  validateAndParseAdParams_(eventData) {
+    this.creativeData = eventData.args.creativeData;
+
+    if (this.creativeData.adParameters == "") {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.UNSPECIFIED, 
+        message: "Ad parameters not found"});
+    }
+
+    let adParams = "";
+    try {
+      adParams = JSON.parse(this.creativeData.adParameters);
+    } catch (exception) {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.CREATIVE_INTERNAL_ERROR, 
+        message: "Invalid JSON input for ad parameters"});
+    }
     const buttonLabel = adParams[AdParamKeys.BUTTON_LABEL]; 
-    //ToDo(juliareichel@): handle case where searchQuery is undefined
     const searchQuery = adParams[AdParamKeys.SEARCH_QUERY];
-    const marker = adParams[AdParamKeys.MARKER];
+
+    if (!searchQuery) {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.UNSPECIFIED, 
+        message: `Required field ${AdParamKeys.SEARCH_QUERY} not found`});
+    }
+    else {
+      super.onInit(eventData);
```

Comment:
super.onInit will call updateInternalOnInit, so I don't think you also need it here.  Alternatively, leave the call here, but just call resolve here instead of super.onINit

Reply:
got it, switched to just calling resolve